### PR TITLE
fix: serial number issue for customer

### PR DESCRIPTION
**Pull request by nehasacher143**
Task: https://app.asana.com/0/1199639423748061/1200681044211594/f

GIF:
Peek 2021-08-05 20-05

### CODE REVIEW

File: erpnext/support/doctype/warranty_claim/warranty_claim.js [hunk 1]
```diff
@@ -15,8 +15,6 @@ frappe.ui.form.on("Warranty Claim", {
 		frm.add_fetch('serial_no', 'maintenance_status', 'warranty_amc_status');
 		frm.add_fetch('serial_no', 'warranty_expiry_date', 'warranty_expiry_date');
 		frm.add_fetch('serial_no', 'amc_expiry_date', 'amc_expiry_date');
-		frm.add_fetch('serial_no', 'customer', 'customer');
```
Comment (hrwX) [inline erpnext/support/doctype/warranty_claim/warranty_claim.js:18]: - Why change something in erpnext when you can always override it via custom app ??

**Review by hrwX — CHANGES_REQUESTED** [review]

File: erpnext/support/doctype/warranty_claim/warranty_claim.js [hunk 2]
```diff
@@ -8,18 +8,33 @@ frappe.ui.form.on("Warranty Claim", {
...
+			return;
+		}
+		frappe.call({
+			"method": "erpnext.support.doctype.warranty_claim.warranty_claim.get_serial_no_values",
+			"args" : {
+				"serial_no": frm.doc.serial_no
+			},
+			callback: function(res) {
+				if(res.message) {
+					frm.set_value(res.message)
```
Comment (hrwX) [inline erpnext/support/doctype/warranty_claim/warranty_claim.js]:
```suggestion
					for (let key in res.message) {
						frm.set_value(key, res.message[key]);
					}
```

File: erpnext/support/doctype/warranty_claim/warranty_claim.js [hunk 3]
```diff
@@ -8,18 +8,33 @@ frappe.ui.form.on("Warranty Claim", {
...
-		frm.add_fetch('serial_no', 'item_name', 'item_name');
-		frm.add_fetch('serial_no', 'description', 'description');
-		frm.add_fetch('serial_no', 'maintenance_status', 'warranty_amc_status');
-		frm.add_fetch('serial_no', 'warranty_expiry_date', 'warranty_expiry_date');
-		frm.add_fetch('serial_no', 'amc_expiry_date', 'amc_expiry_date');
-		frm.add_fetch('serial_no', 'customer', 'customer');
-		frm.add_fetch('serial_no', 'customer_name', 'customer_name');
-		frm.add_fetch('item_code', 'item_name', 'item_name');
-		frm.add_fetch('item_code', 'description', 'description');
+		frm.set_query('serial_no',() => {
```
Comment (hrwX) [inline erpnext/support/doctype/warranty_claim/warranty_claim.js]:
```suggestion
		frm.set_query('serial_no', () => {
```

File: erpnext/support/doctype/warranty_claim/warranty_claim.js [hunk 4]
```diff
@@ -8,18 +8,33 @@ frappe.ui.form.on("Warranty Claim", {
...
+			"method": "erpnext.support.doctype.warranty_claim.warranty_claim.get_serial_no_values",
+			"args" : {
+				"serial_no": frm.doc.serial_no
+			},
+			callback: function(res) {
+				if(res.message) {
+					frm.set_value(res.message)
+				}
+			}
+		})
```
Comment (hrwX) [inline erpnext/support/doctype/warranty_claim/warranty_claim.js]:
```suggestion
		});
```

**Review by hrwX — CHANGES_REQUESTED** [review]

**Comment by sahil28297** [comment]
@nehasacher143 please update your PR with the requested changes and also resolve sider issues.

File: erpnext/support/doctype/warranty_claim/warranty_claim.py [hunk 5]
```diff
@@ -67,3 +67,17 @@ def _update_links(source_doc, target_doc, source_parent):
 			map_child_doc(source_doc, target_doc, table_map, source_doc)
 
 		return target_doc
+
+@frappe.whitelist()
+def get_serial_no_values(serial_no):
+	doc= frappe.get_doc("Serial No", serial_no)
```
Comment (sahil28297) [inline erpnext/support/doctype/warranty_claim/warranty_claim.py]:
```suggestion
	doc = frappe.get_doc("Serial No", serial_no)
```

**Review by sahil28297 — CHANGES_REQUESTED** [review]

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**